x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -157,7 +160,15 @@ public void execute(EsqlQueryRequest request, EsqlExecutionInfo executionInfo, P
             new EsqlSessionCCSUtils.CssPartialErrorsActionListener(executionInfo, listener) {
                 @Override
                 public void onResponse(LogicalPlan analyzedPlan) {
-                    executeOptimizedPlan(request, executionInfo, planRunner, optimizedPlan(analyzedPlan), listener);
+                    try {
+                        queryBuilderResolver.resolveQueryBuilders(
+                            optimizedPlan(analyzedPlan),
+                            listener,
+                            (newPlan, next) -> executeOptimizedPlan(request, executionInfo, planRunner, newPlan, next)
+                        );
+                    } catch (Exception e) {
+                        listener.onFailure(e);
```

Comment:
I'm curious, did you see an Exception here? If so, what type of unchecked exception are we trying to handle. (this is me more trying understand why resolveQueryBuilders is not declaring what it can throw)

Reply:
I was trying to guard against having any type of unhandled exception given that this is an `ActionListener`.